.devcontainer/postcreate.sh

```diff
@@ -0,0 +1,21 @@
+sudo apt update
+sudo apt install -y python3-pip
+
+# Requirements
+pip3 install --user lunr==0.5.5
+pip3 install --user requests
+pip3 install --user pyyaml
+pip3 install --user markdown==2.6.9
+pip3 install --user pygments==2.1.3
+
+# update the site with updated content from external sources/repositories
+python3 update.py --download docs refdoc
+
+# python2 update.py --download codepad examples assets
+
+# the above command fails because of the error:
+
+# Working...Reading classes...Unable to load library 'SpineExt':
+# libSpineExt.so: cannot open shared object file: No such file or directory
+# libSpineExt.so: cannot open shared object file: No such file or directory
+# Native library (linux-aarch64/libSpineExt.so) not found in resource path ([file:/tmp/tmpn2ly7mln/examples-master/build/plugins/defold-spine/plugins/share/pluginSpineExt.jar])
```

Comment:
I could not make `update.py` work for examples. Getting the same error on both local environment and in a devcontainer.

Reply:
This probably works now. There was likely a mismatch at the time with bob version and spine extension version. Could you please try again?

Reply:
The problem was platform (arm64) related. Managed to get it working when forcing amd64 platform on Mac. Now getting a different error when loading "codepad" or "examples":

```
$ python3 update.py --download codepad
Downloading https://github.com/defold/codepad/archive/master.zip to ./codepad-master.zip
Unpacking /tmp/tmpcoepe9v8/codepad-master.zip
Download archive(s)Downloading 0 archive(s)
Working...Reading classes...2024-02-15 01:59:21 INFO     
 1% ...done!
Cleaning... 2% ...done!
Cleaning engine... ...done!
2147483647% ...done!           
Building... 3% 4% 5% 6% 7% 8% 9% 10% 11% 12% 13% 14% 15% 16% 17% 18% 19% 20% 21% 22% 23% 24% 25% 26% 27%2024-02-15 01:59:33 INFO    Compiling main/assets/logo.atlas using profile Default 
 28% 29% 30% 31% 32% 33% 35% 36% 37% 38% 39% 40% 41% 42% 43% 44% 45% 46% 47% 48% 49% 50% 51% 52% 53% 54% 55% 56% 57%java.io.IOException: Problem reading font data.
        at java.desktop/java.awt.Font.createFont0(Font.java:1208)
        at java.desktop/java.awt.Font.createFont(Font.java:1076)
        at com.dynamo.bob.font.Fontc.TTFBuilder(Fontc.java:257)
        at com.dynamo.bob.font.Fontc.compile(Fontc.java:975)
        at com.dynamo.bob.pipeline.GlyphBankBuilder.build(GlyphBankBuilder.java:68)
        at com.dynamo.bob.Project.runTasks(Project.java:1736)
        at com.dynamo.bob.Project.createAndRunTasks(Project.java:1428)
        at com.dynamo.bob.Project.doBuild(Project.java:1539)
        at com.dynamo.bob.Project.build(Project.java:726)
        at com.dynamo.bob.Bob.mainInternal(Bob.java:948)
        at com.dynamo.bob.Bob.main(Bob.java:1008)
 99% ...done!
 100% ...done!

The build failed for the following reasons:
ERROR builtins/fonts/system_font.font:0 Problem reading font data.

Traceback (most recent call last):
  File "/workspaces/defold.github.io/update.py", line 1152, in <module>
    process_codepad(download = args.download)
  File "/workspaces/defold.github.io/update.py", line 647, in process_codepad
    rmcopytree(os.path.join(input_dir, "build", "default", "DefoldCodePad"), codepad_dir)
  File "/workspaces/defold.github.io/update.py", line 105, in rmcopytree
    shutil.copytree(src, dst)
  File "/usr/lib/python3.9/shutil.py", line 555, in copytree
    with os.scandir(src) as itr:
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmpcoepe9v8/codepad-master/build/default/DefoldCodePad'
```

```
$ python3 update.py --download examples
Downloading https://github.com/defold/examples/archive/master.zip to ./examples-master.zip
Processing examples
Unpacking /tmp/tmp95bne9z0/examples-master.zip
..building app
Download archive(s)Downloading 1 archive(s)
 0: Downloading https://github.com/defold/extension-spine/archive/refs/tags/2.13.0.zip
 0: Status 200: Stored /tmp/tmp95bne9z0/examples-master/.internal/lib/e458b3e75af4531d490216575507ceccfdcf394f-ImRkM2M0OTNlZDBhMmM4Y2RlNzAzNzg0ZDMyZGNjNGRmNzRiNTUxZmE0MzQ4ZGMxMmY4OWFjYzg2ZmQ2OTgyYzUi.zip
 2147483647%Working...Reading classes...2024-02-15 02:04:03 INFO    
Found plugins: 
2024-02-15 02:04:03 INFO    Set jna.library.path to '/tmp/tmp95bne9z0/examples-master/build/plugins/defold-spine/plugins/lib/x86_64-linux' 
2024-02-15 02:04:03 INFO    Set java.library.path to '/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib:/tmp/tmp95bne9z0/examples-master/build/plugins/defold-spine/plugins/lib/x86_64-linux' 
2024-02-15 02:04:03 INFO      build/plugins/defold-spine/plugins/share/pluginSpineExt.jar 
2024-02-15 02:04:03 INFO     
 1% ...done!
Reading tasks...Build Remote Engine...
Building engine...Feb 15, 2024 2:04:22 AM com.defold.extender.client.ExtenderClient log
INFO: Sending async build request to https://build.defold.com/build_async/wasm-web/fcec9500fd438e365d4e4f6cec24b99afb907247
Feb 15, 2024 2:04:24 AM com.defold.extender.client.ExtenderClient log
INFO: Async build request was accepted as job job14206322357619828935
Feb 15, 2024 2:04:29 AM com.defold.extender.client.ExtenderClient log
INFO: Waiting for job job14206322357619828935 to finish
2147483647% ...done!           
Building... 2% 3% 4% 5% 6% 7% 8% 9% 10% 11%java.io.IOException: Problem reading font data.
        at java.desktop/java.awt.Font.createFont0(Font.java:1208)
        at java.desktop/java.awt.Font.createFont(Font.java:1076)
        at com.dynamo.bob.font.Fontc.TTFBuilder(Fontc.java:257)
        at com.dynamo.bob.font.Fontc.compile(Fontc.java:975)
        at com.dynamo.bob.pipeline.GlyphBankBuilder.build(GlyphBankBuilder.java:68)
        at com.dynamo.bob.Project.runTasks(Project.java:1736)
        at com.dynamo.bob.Project.createAndRunTasks(Project.java:1428)
        at com.dynamo.bob.Project.doBuild(Project.java:1539)
        at com.dynamo.bob.Project.build(Project.java:726)
        at com.dynamo.bob.Bob.mainInternal(Bob.java:948)
        at com.dynamo.bob.Bob.main(Bob.java:1008)
 99% ...done!
Feb 15, 2024 2:04:39 AM com.defold.extender.client.ExtenderClient log
INFO: Checking job result for job job14206322357619828935
Feb 15, 2024 2:04:40 AM com.defold.extender.client.ExtenderClient log
INFO: Job job14206322357619828935 completed successfully. Writing result to /tmp/build_fcec9500fd438e365d4e4f6cec24b99afb9072478390019170027043532.zip
Feb 15, 2024 2:04:44 AM com.defold.extender.client.ExtenderClient log
INFO: Sending async build request to https://build.defold.com/build_async/js-web/fcec9500fd438e365d4e4f6cec24b99afb907247
Feb 15, 2024 2:04:45 AM com.defold.extender.client.ExtenderClient log
INFO: Async build request was accepted as job job13677299320465678528
Feb 15, 2024 2:04:50 AM com.defold.extender.client.ExtenderClient log
INFO: Waiting for job job13677299320465678528 to finish
Feb 15, 2024 2:05:16 AM com.defold.extender.client.ExtenderClient log
INFO: Checking job result for job job13677299320465678528
Feb 15, 2024 2:05:16 AM com.defold.extender.client.ExtenderClient log
INFO: Job job13677299320465678528 completed successfully. Writing result to /tmp/build_fcec9500fd438e365d4e4f6cec24b99afb90724711036580825127347925.zip
 ...done!
2024-02-15 02:05:17 INFO    Engine build took 65.487000 s 
 100% ...done!

The build failed for the following reasons:
ERROR assets/text64.font:0 Problem reading font data.

...copying app
Traceback (most recent call last):
  File "/workspaces/defold.github.io/update.py", line 1144, in <module>
    process_examples(download = args.download)
  File "/workspaces/defold.github.io/update.py", line 558, in process_examples
    rmcopytree(os.path.join(input_dir, "build", "default", "Defold-examples", "archive"), os.path.join(examples_dir, "archive"))
  File "/workspaces/defold.github.io/update.py", line 105, in rmcopytree
    shutil.copytree(src, dst)
  File "/usr/lib/python3.9/shutil.py", line 555, in copytree
    with os.scandir(src) as itr:
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmp95bne9z0/examples-master/build/default/Defold-examples/archive'
```

Reply:
We did a change to the font data format recently where font glyph banks were moved to their own data structure. It might be the reason why you are getting this problem? Which version of bob.jar is used?

Reply:
This one http://d.defold.com/archive/fcec9500fd438e365d4e4f6cec24b99afb907247/bob/bob.jar

Reply:
Strange. Can you reproduce that font issue outside of the dev container with the same set of tools installed and with those exact versions?

Reply:
The problem's solved.

<img width="1107" alt="image" src="https://github.com/defold/defold.github.io/assets/7230306/97514d2e-d80a-421b-94e5-d8e83a8e7427">